client/modules/IDE/components/FileNode.jsx

```diff
@@ -10,6 +10,47 @@ import FolderRightIcon from '../../../images/triangle-arrow-right.svg';
 import FolderDownIcon from '../../../images/triangle-arrow-down.svg';
 import FileIcon from '../../../images/file.svg';
 
+function FileName({ name }) {
+  const nameArray = name.split('.');
```

Comment:
It's a good idea to catch `if (!name) return;` at the beginning of the function, otherwise `.split` can explode

Reply:
`name` is a required string proptype, which means that the component itself will complain if it's used without this prop, so is this necessary?

Reply:
True, in this case it's kind of pointless